### PR TITLE
Support for trial extension coupons

### PR DESCRIPTION
**Pull request by bhelx**
This feature is in production but is not available for every site yet. I've added the feature to our sandbox site. It's supposed to be available on Thursday.

The feature is a special type of coupon that allows an extension of your subscription trial period.

The API changes are only a few added fields.

\cc @pierre

### CODE REVIEW

File: src/test/java/com/ning/billing/recurly/TestRecurlyClient.java [hunk 1]
```diff
@@ -1112,4 +1112,45 @@ public void testRedeemCoupon() throws Exception {
...
+        couponData.setName("apitrialext");
+        couponData.setCouponCode("apitrialext");
+        couponData.setTrialExtensionAmount(3);
+        couponData.setTrialExtensionUnit("month");
+        couponData.setDiscountType("trial_extension");
+
+        try {
+            final Account account = recurlyClient.createAccount(accountData);
+            final Plan plan = recurlyClient.createPlan(planData);
+            final Coupon coupon = recurlyClient.createCoupon(couponData);
```
Comment (pierre) [inline src/test/java/com/ning/billing/recurly/TestRecurlyClient.java]: Is the coupon automatically redeemed? Or should it be applied to the subscription explicitly?

Comment (bhelx) [inline src/test/java/com/ning/billing/recurly/TestRecurlyClient.java]: The `coupon_code` needs to be added to the `subscriptionData` below. The test should then check that the coupon was applied. I'll update this when the feature is ready :)

**Comment by bhelx** [comment]
@pierre Can you give this another look when you get the chance? The feature is in production now.

File: src/test/java/com/ning/billing/recurly/TestRecurlyClient.java [hunk 2]
```diff
@@ -1112,4 +1113,50 @@ public void testRedeemCoupon() throws Exception {
...
+            subscriptionData.setCouponCode(coupon.getCouponCode());
+
+            subscriptionData.setPlanCode(plan.getPlanCode());
+            subscriptionData.setAccount(accountData);
+            subscriptionData.setCurrency(CURRENCY);
+            subscriptionData.setUnitAmountInCents(1242);
+
+            final Subscription subscription = recurlyClient.createSubscription(subscriptionData);
+            Assert.assertNotNull(subscription);
+            Assert.assertEquals(subscription.getTrialEndsAt().getMonthOfYear(), now.getMonthOfYear() + 3);
```
Comment (bhelx) [inline src/test/java/com/ning/billing/recurly/TestRecurlyClient.java]: This is meant to check that the trial was extended by 3 months

**Comment by pierre** [comment]
👍

**Pull request merged**